Src/PerformanceCollector/Unit.Tests/QuickPulse/Mocks/ClockMock.cs

```diff
@@ -6,19 +6,27 @@
 
     internal class ClockMock : Clock
     {
+        private readonly object lockObj = new object();
+
         private DateTimeOffset now = DateTimeOffset.UtcNow;
 
         public override DateTimeOffset UtcNow
         {
             get
             {
-                return this.now;
+                lock (this.lockObj)
```

Comment:
nitpick: Since this is a test it probably doesn't really matter but you could avoid the lock on the getter if you did an interlocked reference exchange in the setter.